### PR TITLE
Fluid rework derp fixes.

### PR DESCRIPTION
**Pull request by gigaherz**
As expected, there's some things I completely missed before merging. This is what I have found so far. Feel free to poke at me with anything I may have missed.

### CODE REVIEW

**Comment by Shadows-of-Fire** [comment]
With reference to FluidStacks:  
Personally I think all of the checking to ensure that the amount is >=0 are overkill.  Once the amount is <= 0 the stack should be empty and the amount should no longer be relevant.  If anything the setAmount call should clamp at 0, but not crash.
Additionally, why does it matter if a user edits the size of the empty fluid stack?  It will always be empty regardless of size.

**Comment by gigaherz** [comment]
The check exists as a sanity test. If a mod is setting the amount of an empty stack, their code is broken, so it is better to provide a clear indication of that fact.

Wrt. getAmount checks, I could have missed some. My intention was that all cases of getAmount() > 0 or <= 0 were converted into isEmpty(). But my primary concern were the nulls, so the `getAmount()` calls slipped off my mind.

**Comment by Shadows-of-Fire** [comment]
That was more with relation to the number of checks being done in setAmount.  Might be insignificant but also unnecessary.

**Comment by gigaherz** [comment]
I would disagree on the unnecessary part. I think it is an important sanity check.

To clarify: I'm willing to remove that check if it is a performance concern, but I would rather keep it if the performance hit is insignificant.

File: src/test/java/net/minecraftforge/debug/fluid/NewFluidTest.java [hunk 1]
```diff
@@ -103,6 +103,16 @@ public void registerFluids(RegistryEvent.Register<Fluid> event)
         event.getRegistry().registerAll(test_fluid, test_fluid_flowing);
     }
 
+    public void loadComplete(FMLLoadCompleteEvent event)
+    {
+        // some sanity checks
+        BlockState state = Fluids.WATER.getDefaultState().getBlockState();
+        BlockState state2 = Fluids.WATER.getAttributes().getBlock(null,null,Fluids.WATER.getDefaultState());
+        assert state.getBlock() == Blocks.WATER && state2 == state;
```
Comment (ichttt) [inline src/test/java/net/minecraftforge/debug/fluid/NewFluidTest.java]: Please use Validate.isTrue instead of assert, no one enables assertions

Comment (gigaherz) [reply src/test/java/net/minecraftforge/debug/fluid/NewFluidTest.java]: Right.

**Comment by gigaherz** [comment]
Merging this to help people have a more stable base to test on.

**Pull request merged**